### PR TITLE
DAOS-623 test: Remove some suppressions in tests and fix leaks

### PR DESCRIPTION
**Pull request by jolivier23**
Fix some memory leaks and remove suppressions
Utilize D_HAS_VALGRIND to reduce runtime for memcheck
Move btree perf tests out of valgrind set and reduce size under valgrind
Use D_* allocation routines in eq_tests consistently

Signed-off-by: Jeff Olivier <jeffrey.v.olivier@intel.com>
Co-authored-by: Kris Jacque <kristin.jacque@intel.com>

### CODE REVIEW

**Comment by ashleypittman** [comment]
Did you mean to request reviews on this rather than assign it to us?

**Comment by jolivier23** [comment]
> Did you mean to request reviews on this rather than assign it to us?

Yeah, sorry.  I saw it passed tests on my phone and did that on accident.  Fixed

**Review by ashleypittman — APPROVED** [review]

**Review by mchaarawi — APPROVED** [review]

File: src/security/cli_security.c [hunk 1]
```diff
@@ -33,6 +33,7 @@ dc_sec_request_creds(d_iov_t *creds)
 
 	rc = request_credentials_via_drpc(&response);
 	if (rc != DER_SUCCESS) {
+		drpc_response_free(response);
```
Comment (kjacque) [inline src/security/cli_security.c:36]: Doesn't hurt anything, but I suspect what you actually ran into here was an issue in a test, not an issue in the production code. In practice drpc_call shouldn't be allocating a response if it returned a failure. But at this level we're mocking drpc_call in the unit tests, so the test probably didn't clean up the mock.

File: utils/run_test.sh [hunk 2]
```diff
@@ -98,6 +98,13 @@ if [ -d "/mnt/daos" ]; then
         export DAOS_IO_BYPASS=pm_snap
         run_test "${SL_PREFIX}/bin/vos_tests" -A 50
         unset DAOS_IO_BYPASS
+        COMP="UTEST_common"
+        run_test src/common/tests/btree.sh perf -s 20000
+        run_test src/common/tests/btree.sh perf direct -s 20000
+        run_test src/common/tests/btree.sh perf ukey -s 20000
+        run_test src/common/tests/btree.sh dyn perf -s 20000
+        run_test src/common/tests/btree.sh dyn perf ukey -s 20000
+        BTREE_SIZE=20000
```
Comment (kjacque) [inline utils/run_test.sh:107]: Might be better to set BTREE_SIZE first and use the variable when calling the tests

**Review by kjacque — COMMENTED** [review]
I see that this is merged already, just had a couple minor comments.

**Pull request merged**